satpy/readers/seviri_l1b_hrit.py

```diff
@@ -476,7 +368,7 @@ def _get_header(self):
         self.mda['projection_parameters']['SSP_latitude'] = 0.0
 
         # Orbital parameters
-        actual_lon, actual_lat, actual_alt = self.prologue_.get_satpos()
+        actual_lon, actual_lat, actual_alt = self.prologue_.satpos
```

Comment:
I understand from the code that `satpos` will be `(None, None, None)` there is not valid polynomial, since the original `get_satpos` raises an exception which is intercepted by `get_satpos_safe`.

Now, I would rather remove `get_satpos_safe` and just get the exception.
In this `__init__` for example, that would mean catching the exception instead of checking for `actual_*` being `None`, which I think would explain more why the `orbital_parameters` aren't set, especially since the name of the exception is so explicit (`NoValidOrbitParams`)

Reply:
This is highly subjective of course, feel free to argue.

Reply:
Yes, that's a good solution! The `get_satpos_safe` didn't feel right, but I didn't come up with a better solution (to avoid catching and logging the exception in each reader). But since I check for `None` type satellite positions in each reader anyway, it's more clear to catch the exception instead. Thanks for the hint!